saiph/reduction/famd.py

```diff
@@ -532,3 +531,84 @@ def _compute_cos2_single_category(
         np.array(cos2) / summed_weights_without_zeros
     )
     return single_category_cos2
+
+
+def reconstruct_df_from_model(model: Model) -> pd.DataFrame:
+    """Reconstruct a DataFrame from a fitted model.
+
+    Note: if nf < df.shape[1], reconstructed df will not be exactly the same.
+
+    Parameters:
+        model: Model computed by fit.
+
+    Returns:
+        df: The reconstructed DataFrame.
+    """
+    # Step 1: Extract the necessary components from the model
+    U = model.U
+    if model.s is None:
+        raise ValueError(
+            "Model has not been fitted. Call fit() to create a Model instance."
+        )
+    S = model.s
+    Vt = model.V
+    row_w = model.row_weights
+    col_weights = model.column_weights
+    if model.mean is None:
+        raise ValueError(
+            "Model has not been fitted. Call fit() to create a Model instance."
+        )
```

Comment:
already done before, is it necessary? if yes, prefer
```
if cond_1 or cond_2 or cond_3 ... : 
	raise blablabla
```

Reply:
hum I guess it is needed for mypy I suppose, so I think the suggestion above should work

Reply:
Yep Only for mypy as depending on mca, pca, or famd, some elements are optional.
Testing this seems useless as the model can't be obtained by the user unless he fits